Common/Util/Validate.cs

```diff
@@ -0,0 +1,106 @@
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+*/
+
+using System;
+using System.Globalization;
+using System.Text.RegularExpressions;
+
+namespace QuantConnect.Util
+{
+    /// <summary>
+    /// Provides methods for validating strings following a certain format, such as an email address
+    /// </summary>
+    public static class Validate
+    {
+        /// <summary>
+        /// Validates the provided email address
+        /// </summary>
+        /// <remarks>
+        /// Implementation taken from msdn (with slight refactoring for readability and C#6 compliance):
+        /// https://docs.microsoft.com/en-us/dotnet/standard/base-types/how-to-verify-that-strings-are-in-valid-email-format
+        /// </remarks>
+        /// <param name="emailAddress">The email address to be validated</param>
+        /// <returns>True if the provided email address is valid</returns>
+        public static bool EmailAddress(string emailAddress)
+        {
+            if (string.IsNullOrWhiteSpace(emailAddress))
+            {
+                return false;
+            }
+
+            emailAddress = NormalizeEmailAddressDomainName(emailAddress);
+            if (emailAddress == null)
+            {
+                // an error occurred during domain name normalization
+                return false;
+            }
+
+            try
+            {
+                return RegularExpression.Email.IsMatch(emailAddress);
+            }
+            catch
+            {
+                return false;
+            }
+        }
+
+        private static string NormalizeEmailAddressDomainName(string emailAddress)
+        {
+            try
+            {
+                // Normalize the domain
+                emailAddress = RegularExpression.EmailDomainName.Replace(emailAddress, match =>
+                {
+                    // Use IdnMapping class to convert Unicode domain names.
+                    var idn = new IdnMapping();
+
+                    // Pull out and process domain name (throws ArgumentException on invalid)
+                    var domainName = idn.GetAscii(match.Groups[2].Value);
+
+                    return match.Groups[1].Value + domainName;
+                });
+            }
+            catch
+            {
+                return null;
+            }
+
+            return emailAddress;
+        }
+
+        /// <summary>
+        /// Provides static storage of compiled regular expressions to preclude parsing on each invocation
+        /// </summary>
+        public static class RegularExpression
```

Comment:
nit: wondering if it's worth for this guy to be public and not private, thoughts?

Reply:
I had gone back and forth on this one and figured exposing it doesn't do any harm. The added benefit to exposing it is there can be capture groups in the regular expressions that can be useful depending on what you're doing. We would need to expose methods for fetching the capture groups for each regex. This behavior isn't currently used, but was the reasoning. I'm happy to move it to private as well.

Reply:
Agree, there is no one right choice here, personally, I'd keep it private until required but it's good as it too 👍